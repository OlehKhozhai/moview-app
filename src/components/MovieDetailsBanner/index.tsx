import React from 'react';
import cn from 'classnames';

import Button from 'components/_common/Button';
import Logo from 'components/_common/Logo';
import styles from './styles.module.scss';
import { Movie } from 'redux/types';

type MovieDetailsBannerProps = {
  movie: Movie;
  className?: string;
};

const MovieDetailsBanner: React.FC<MovieDetailsBannerProps> = ({ movie, className }) => {
  return (
    <div className={cn(styles.root, className)}>
      <header className={styles.header}>
        <Logo />
        <Button title="Search" linkTo="/" />
      </header>

      <div className={styles.content}>
        <div className={styles.imageWrapper}>
          <img className={styles.image} src={movie.poster_path} />
        </div>

        <div>
          <h2 className={styles.title}>{movie.title}</h2>
          <span className={styles.genre}>
            {movie.genres.map((genre) => (
              <span key={genre}> {genre} |</span>
            ))}
          </span>

          <div className={styles.warnRow}>
            <span className={styles.year}>{movie.release_date}</span>
            <span>{movie.runtime}</span>
          </div>

          <p className={styles.description}>{movie.overview}</p>
        </div>
      </div>
    </div>
  );
};

export default React.memo(MovieDetailsBanner);
