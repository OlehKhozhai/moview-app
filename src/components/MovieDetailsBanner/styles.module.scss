@import 'assets/styles/mixins/colors';

.root {
  padding: 10px 60px 60px;
  background-color: $primary-bg;
}

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 20px;
}

.content {
  display: flex;
  color: $primary-color;
}

.imageWrapper {
  max-width: 280px;
  width: 100%;
  margin-right: 60px;
}

.image {
  width: 100%;
  object-fit: cover;
}

.title {
  text-transform: uppercase;
  font-size: 40px;
  margin: 50px 0 20px 0;
}

.genre {
  color: $primary-dark-color;
}

.warnRow {
  color: $secondary-color;
  margin: 20px 0 35px 0;
  font-size: 30px;
}

.year {
  margin-right: 40px;
}

.description {
  line-height: 1.5;
}
