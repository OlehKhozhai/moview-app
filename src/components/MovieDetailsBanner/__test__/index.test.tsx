import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import MovieDetailsBanner from '../index';

describe('MovieDetailsBanner component', () => {
  test('should render MovieDetailsBanner component by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <MovieDetailsBanner
          movie={{
            id: 1,
            budget: 100,
            genres: ['Documentary'],
            overview: 'Overview',
            poster_path: 'https://via.placeholder.com/330x450',
            title: 'Movie 1',
            release_date: '1999',
            revenue: 102,
            runtime: 12312,
            tagline: 'tagline',
            vote_average: 4,
            vote_count: 100,
          }}
        />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
