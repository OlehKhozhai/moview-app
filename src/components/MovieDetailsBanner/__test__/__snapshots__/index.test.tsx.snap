// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsBanner component should render MovieDetailsBanner component by default 1`] = `
<div
  className="root"
>
  <header
    className="header"
  >
    <a
      className="root"
      href="/"
      onClick={[Function]}
    >
      <span
        className="bold"
      >
        netflix
      </span>
      roulete
    </a>
    <a
      className="root primary"
      href="/"
      onClick={[Function]}
    >
      Search
    </a>
  </header>
  <div
    className="content"
  >
    <div
      className="imageWrapper"
    >
      <img
        className="image"
        src="https://via.placeholder.com/330x450"
      />
    </div>
    <div>
      <h2
        className="title"
      >
        Movie 1
      </h2>
      <span
        className="genre"
      >
        <span>
           
          Documentary
           |
        </span>
      </span>
      <div
        className="warnRow"
      >
        <span
          className="year"
        >
          1999
        </span>
        <span>
          12312
        </span>
      </div>
      <p
        className="description"
      >
        Overview
      </p>
    </div>
  </div>
</div>
`;
